lib/ansible/module_utils/common/file.py

```diff
@@ -49,57 +49,57 @@ class FileLock:
     '''
     Currently FileLock is implemented via fcntl.flock on a lock file, however this
     behaviour may change in the future. Avoid mixing lock types fcntl.flock,
-    fcntl.lockf and module_utils.common.file.FileLock as it will certainly cause
+    fcntl.flock and module_utils.common.file.FileLock as it will certainly cause
     unwanted and/or unexpected behaviour
     '''
     def __init__(self):
         self.lockfd = None
 
     @contextmanager
-    def lock_file(self, path, tmpdir, lock_timeout=None):
+    def lock_file(self, path, lock_timeout=None):
         '''
         Context for lock acquisition
         '''
         try:
-            self.set_lock(path, tmpdir, lock_timeout)
+            self.set_lock(path, lock_timeout)
             yield
         finally:
             self.unlock()
 
-    def set_lock(self, path, tmpdir, lock_timeout=None):
+    def set_lock(self, path, lock_timeout=None):
         '''
-        Create a lock file based on path with flock to prevent other processes
-        using given path.
-        Please note that currently file locking only works when it's executed by
-        the same user, I.E single user scenarios
+        Set lock on given path via fcntl.flock(), note that using
+        locks does not guarantee exclusiveness unless all accessing
+        processes honor locks.
 
         :kw path: Path (file) to lock
-        :kw tmpdir: Path where to place the temporary .lock file
         :kw lock_timeout:
             Wait n seconds for lock acquisition, fail if timeout is reached.
             0 = Do not wait, fail if lock cannot be acquired immediately,
             Default is None, wait indefinitely until lock is released.
         :returns: True
         '''
-        lock_path = os.path.join(tmpdir, 'ansible-{0}.lock'.format(os.path.basename(path)))
+        b_lock_path = to_bytes(path, errors='surrogate_or_strict')
         l_wait = 0.1
         r_exception = IOError
         if sys.version_info[0] == 3:
             r_exception = BlockingIOError
 
-        self.lockfd = open(lock_path, 'w')
+        self.lockfd = open(b_lock_path, 'ab')
```

Comment:
We'll need to change this to raise an IOError if the file does not exist.  With the previous code, this was okay because we were opening and locking a temporary file but now that we're operating on the real file we need to ensure that we don't end up creating a nonexistent file in check mode.